runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -153,9 +155,32 @@ public final class Engine implements AutoCloseable
         final Map<String, Guard> guardsByType = guards.stream()
             .collect(Collectors.toMap(g -> g.name(), g -> g));
 
-        final Callable<Void> configure =
-                new ConfigureTask(configURL, schemaTypes, guardsByType::get, labels::supplyLabelId, maxWorkers, tuning,
-                        dispatchers, errorHandler, logger, context, config, extensions);
+
+        registerTask = new RegisterTask(configURL, schemaTypes, guardsByType::get,
+            labels::supplyLabelId, maxWorkers, tuning, dispatchers, errorHandler, logger, context, config, extensions);
+
+        if (configURL == null || "file".equals(configURL.getProtocol()))
```

Comment:
Can `configURL` actually be null?

Reply:
Originally, it could be null in ITs. But I changed so it can't be by setting it to the expected URL, so we can watch config file change for the 0-1 (create) scenario.

Reply:
So you're right. I can delete this check